src/mesh/boundary_info.C

```diff
@@ -259,6 +292,15 @@ void BoundaryInfo::sync (const std::set<boundary_id_type> &requested_boundary_id
   // and finally distribute element partitioning to the nodes
   Partitioner::set_node_processor_ids(boundary_mesh);
 
+  if(node_id_map_ptr)
+  {
+    *node_id_map_ptr = node_id_map;
+  }
```

Comment:
Pretty inefficient to make a second copy right before destroying the first, no?  Do a swap() instead?